product_category_company_favorite/models/product_category.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2023-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class ProductCategory(models.Model):
+    _inherit = "product.category"
+
+    is_favorite = fields.Boolean(
+        string="Favorite",
+        company_dependent=True,
+        help="If this field is unchecked, the category will"
+        " be hidden when searching category in a drop-down list"
+        " like in the product form view."
+    )
+
+    @api.onchange("parent_id")
+    def onchange_parent_id_favorite(self):
+        if self.parent_id:
+            self.is_favorite = self.parent_id.is_favorite
+
+    @api.model
+    def create(self, vals):
+        category = super().create(vals)
+        # Set the new category as a favorite for all the other companies
+        company_ids = self.env["res.company"].with_context(
+            active_test=False
+        ).search([("id", "!=", self.env.user.company_id.id)]).ids
+        for company_id in company_ids:
+            contextual_category = category.with_context(force_company=company_id)
+            contextual_category.is_favorite = (
+                not contextual_category.parent_id
+                and True
+                or contextual_category.is_favorite
+            )
```

Comment:
Hard to read.

```suggestion
            contextual_category.is_favorite = (
                True if not contextual_category.parent_id
                else contextual_category.is_favorite
            )
```

I'm not really sure what `else contextual_category.is_favorite` means here, though. I guess it defaults to False? Why not be explicit?

```suggestion
            contextual_category.is_favorite = not bool(
                contextual_category.parent_id
            )
```

Reply:
I think I mean or ``contextual_category.parent_id.is_favorite`` to make the feature similar to the onchange.

If we create a new root category, by default it is available in other companies.
If we create a new child category, by default, it inherit from the configuration of it parent in other companies.

does it make sense for you ?

Reply:
> I'm not really sure what else contextual_category.is_favorite means here, though. I guess it defaults to False? Why not be explicit?

Reading again : 
- you're right It makes no sense. 
- I changed the algorithm. 
  - if it is a child category -> inheritance : OK
  - if it is a new root category : same configuration as the current company

So it allow to create "opt-in" categories, or "opt-out" categories, depending on the user.